sdk/src/trace/span.cc

```diff
@@ -94,7 +95,7 @@ Span::Span(std::shared_ptr<Tracer> &&tracer,
   }
 
   span_context_ = std::unique_ptr<trace_api::SpanContext>(
-      new trace_api::SpanContext(trace_id, span_id, trace_api::TraceFlags(), false,
+      new trace_api::SpanContext(trace_id, span_id, trace_flags, false,
```

Comment:
Not sure if I am missing something here - Why do we need to copy the parent's trace_flags to child span. For sampling-bit in child span trace_flag it should be determined as below - 
- Parent trace_flags MAY be used for sampling decision to decide whether child span should be sampled.
- In case the child span should be sampled ( as decided by above sampling algo ) - we should set the sampling-bit in child span trace_flag.

Or is there more information in parent's trace_flag ( other than sampling-bit) which we want to copy from parent ? If that is the case, we should copy that  first and then overwrite sampling bit based on decision from above logic.

Though our current implementation is also wrong - as it's assigning 0x00 to trace_flags even if it's is supposed to be sampled. This should be fixed.

Reply:
Do you mean we set `kIsSampled` directly in the above code line because the location implies sampling should be enabled, instead of copying `trace_flags` from parent `SpanContext`? If so, I think we could keep the same sampled value as parent to be consistent and leave the decision to sampler, like even the IsSampled is off, some sampler may still sample it anyway. But I am fine to set the sampled bit here too.

Reply:
If it helps, how it is done in JS:
https://github.com/open-telemetry/opentelemetry-js/blob/71ba83a0dc51118e08e3148c788b81fe711003e7/packages/opentelemetry-tracing/src/Tracer.ts#L96

and python:

https://github.com/open-telemetry/opentelemetry-python/blob/fb1ae06026d648d867461396a6d1b5c37cbb3a31/opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py#L924

Reply:
> Do you mean we set kIsSampled directly in the above code line because the location implies sampling should be enabled, instead of copying trace_flags from parent SpanContext?

Yes, as we are creating the span (and sampling decision happens before that in tracer), that means decision has already been made by sampler to sample it, and we should just be setting sampling-bit in the span's trace_flag irrespective of what is there in parent's trace_flag.

Reply:
@lalitb thanks for the links. Set sampled based sampler decision makes sense. Updated the PR accordingly.